desc/optimize/_constraint_wrappers.py

```diff
@@ -142,34 +126,38 @@ def recover(self, x_reduced):
         """Recover the full state vector from the reduced optimization vector."""
         return self._recover(x_reduced)
 
-    def x(self, eq):
+    def x(self, *things):
         """Return the reduced state vector from the Equilibrium eq."""
-        x = np.zeros((self._dim_x_full,))
-        for arg in self.args:
-            x[self.x_idx[arg]] = getattr(eq, arg)
+        x = self._objective.x(*things)
         return self.project(x)
 
-    def unpack_state(self, x):
+    def unpack_state(self, x, per_objective=True):
         """Unpack the state vector into its components.
 
         Parameters
         ----------
         x : ndarray
             State vector, either full or reduced.
+        per_objective : bool
+            Whether to return param dicts for each objective (default) or for each
+            unique optimizable thing.
```

Comment:
What is the use case for each option?

Reply:
basically depends what you're iterating over. IE, for most optimization stuff you want to get the params that will be passed to each objective. But at the end when you're assigning the optimized parameters, you want a single dict for each optimizable object.